apis/destination.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"net/url"
+	"strings"
+)
+
+// Destination represents a target of an invocation over HTTP.
+type Destination struct {
+	// ObjectReference points to an Addressable.
+	*corev1.ObjectReference `json:",inline"`
+
+	// URI is for direct URI Designations.
+	URI *URL `json:"uri,omitempty"`
+
+	// Path is used with the resulting URL from Addressable ObjectReference or
+	// URI. Must start with `/`.
+	Path *string `json:"path,omitempty"`
```

Comment:
Does this override the path in the resolved URL or augment it?

Reply:
I added a comment that the intent is to append them but it is up to the implementation of the resolver. Which will be added to pkg in a followup.

Reply:
Having the semantics be context specific feels somewhat antithetical to duck typing, does it not?

Reply:
This is not a duck, it is a field implementation.